index.js

```diff
@@ -0,0 +1,77 @@
+
+const express = require("express");
+const shortid = require("shortid"); 
+
+const server = express();
+
+let users = [];
+
+server.use(express.json());
+
+
+//GET THE ARRAY OF USERS
+server.get("/users", (req, res) => {
+// const users = [
+//     {
+//         id: 1,
+//         username: 'Bob'
+//     },
+//     {
+//         id: 2,
+//         username: 'Mercutio'
+//     }]
+   res.status(200).json(users)
+});
+
+
+//GET A SINGLE USER BY ID
+server.get("/users/:id", (req, res) => {
+
+    users.map(user => {
+        if(user.id === req.body.id){
```

Comment:
req.params.id is the id that's in the url. For example /users/3. We want to find by comparing the id in the url and the user.id